### PR TITLE
 Add option to filter TestCases by date

### PR DESCRIPTION
**Pull request by asankov**
Refs #123, #715

I have modeled this after `testplans/search.html` where we have a similar selector. The `initDateTimePicker` function is a good candidate for `utils.js` so we can reuse it in both places, currently we have the same code inline in the `testplans/search.html` template. I can do it here, or in a separate PR. 

I went through the translations diff and there are no commented or fuzzy ones. However, I have no idea why `manage.py makemessages` is generating such a large diff, most of the changes are just changes of the line, where this translation is used, but it is totally unrelated to my changes.
Also, we have diffs like this:
```diff
-- msgid "Updated on %(history_date)s\n"
++ msgid ""
++ "Updated by %(username)s\n\n"
```
Is it something OS specific (I work on Mac) ? Can you advice how to handle this (staging chunk does not always work)?

### CODE REVIEW

**Comment by atodorov** [comment]
>  went through the translations diff and there are no commented or fuzzy ones. However, I have no idea why manage.py makemessages is generating such a large diff,

Because there's a difference in how Django and Crowdin work with larger lines of text. In addition Django isn't smart enough to not recognized there are no string changes and the issue I found about this was WONTFIX because the initial gettext specification isn't quite clear about it.

Unless we manage to get makemessages patched and approved by the Django community this will remain PITA.

File: tcms/testcases/static/testcases/js/search.js [hunk 1]
```diff
@@ -124,3 +135,15 @@ $(document).ready(function() {
 
     $('.selectpicker').selectpicker();
 });
+
+function initDateTimePicker(selector) {
+    $(selector).datetimepicker({
+        locale: '{{ LANGUAGE_CODE }}',
```
Comment (atodorov) [inline tcms/testcases/static/testcases/js/search.js]: There is a reason this function is in the HTML template. Variable substitution will not work in JavaScript b/c this doesn't go through Django's templating engine.

Comment (atodorov) [reply tcms/testcases/static/testcases/js/search.js]: You can move this into a separate HTML template and include it where necessary. Also moving this function into utils.js

**Comment by asankov** [comment]
Build seems to be failing with the following error:
```
ModuleNotFoundError: No module named 'OpenSSL'
make: *** [build-for-pypi] Error 1
```
Please advise on what to do. I don't think any of my changes affects this

**Comment by atodorov** [comment]
> ModuleNotFoundError: No module named 'OpenSSL'

rebase, this is coming from the latest versions of twine which we use during testing

**Comment by atodorov** [comment]
Cherry-picked in #759

**Pull request closed without merging**